frontend/src/utilities/useDraggableTable.ts

```diff
@@ -0,0 +1,162 @@
+import * as React from 'react';
+import { TbodyProps, TrProps } from '@patternfly/react-table';
+import styles from '@patternfly/react-styles/css/components/Table/table';
+import useDebounce from '~/utilities/useDebounce';
+
+export type TrDragFunctionsType = {
+  onDragStart: React.DragEventHandler<HTMLTableRowElement>;
+  onDragEnd: React.DragEventHandler<HTMLTableRowElement>;
+  onDrop: React.DragEventHandler<HTMLTableRowElement>;
+};
+
+const useDraggableTable = (
+  bodyRef: React.RefObject<HTMLTableSectionElement>,
+  initialItemOrder: string[],
+) => {
+  const [draggedItemId, setDraggedItemId] = React.useState('');
+  const [draggingToItemIndex, setDraggingToItemIndex] = React.useState(-1);
+  const [isDragging, setIsDragging] = React.useState(false);
+  const [itemOrder, setItemOrder] = React.useState(initialItemOrder);
+  const [tempItemOrder, setTempItemOrder] = React.useState<string[]>([]);
+
+  const debouncedItemOrder = useDebounce(itemOrder, 5000);
```

Comment:
This 5 secs debounce might be overkill, I might wanna have feedback from UX here, @vconzola 

This is the "lag" between a sort action and the dashboard committing the changes in openshift (to have persistence). In this scenario, if a user rearranges the table and then navigates to other screen saves will be lost. 

Not sure if we wanna decrease the time here.

Reply:
We can solve this later, everything else seems fine, so we might wanna reciew it later